### ISSUE TITLE
ClassExpand() to include leading separator

### ISSUE THREAD

**Issue by FatBoyXPC** [phpactor/phpactor#366]
Right now the very first `\` is left off of the string when calling `ClassExpand()`. The only time I try to expand a class is when it's inside a function signature or a DocBlock (as opposed to a `use` statement where it would make sense to leave the `\` off). Maybe `ClassExpand()` could take an argument for whether or not to put the leading `\`?

**Comment by dantleech** [comment]
It makes sense to **NOT** add the `\` on a `use Foobar`. However not so much when in a class scope.

**Comment by FatBoyXPC** [comment]
If making this behave right in the various uses cases is annoying or difficult, I'd honestly be okay with the argument option so I could at least put it in my vimrc and forget about it

**Comment by RadusiViorel** [comment]
This issue is a pain if you want to use the LSP autocomplete from roxma/LanguageServer-php-neovim, without the leading separator it will not find the class.

**Comment by FatBoyXPC** [comment]
That's not an issue with autocomplete if you use [coc.nvim](https://github.com/neoclide/coc.nvim), that said, the expand class feature putting it in docblocks would still be nice :D

**Comment by dantleech** [comment]
> without the leading separator it will not find the class.

I only expand classes in docblocks / xml / yaml etc. How does class expansion relate to auto-completion?

**Comment by RadusiViorel** [comment]
I have 3 screenshot to show the case:
1. Here in the docblocks the `InputInterface` path doesn't start with the `\` ,  so when  i type `->` nothing will happen.
![Screen Capture_select-area_20200605105039](https://user-images.githubusercontent.com/20063631/83851359-e9b28300-a71a-11ea-8db8-43e5185569ba.png)

2. For the `OutputInterface` i added the`\` and things start working as expected
![Screen Capture_select-area_20200605105056](https://user-images.githubusercontent.com/20063631/83851496-21b9c600-a71b-11ea-9dca-ebfc135d3380.png)
![Screen Capture_select-area_20200605105201](https://user-images.githubusercontent.com/20063631/83851500-22eaf300-a71b-11ea-94a7-d2dcf6de2fba.png)

And i do use [coc.nvim](https://github.com/neoclide/coc.nvim) to start the [SLP plugin](https://github.com/felixfbecker/php-language-server) ( still i am new to this all of this so maybe i am missing something) . I just added the lines need to start the server in  `:CocConfig` 
```
  "languageserver": {
    "phplang": {
      "command": "php",
      "args": ["/path/to/vendor/felixfbecker/language-server/bin/php-language-server.php"],
      "filetypes": ["php"]
    }
  ```

**Comment by RadusiViorel** [comment]
Ok , sorry. I just found out the the autocomplete shown above is not provider by this  [SLP plugin ](https://github.com/felixfbecker/php-language-server). I stopped the server and the behavior is the same, not sure what plugin provides it ,i think it's nmc2-phpactor.

**Comment by dantleech** [comment]
Actually the autocomplete is working as expected. Without the leading `\` it is relative to the current namespace (so `Your\\Namespace\\Blah\\Symfony\\Componsen\\Console\\InputInterface` would be resolved or similar).

Phactor could be smarted about this indeed.

But I'd also suggestion avoiding these docblocks entirely. The information is already in the type hints :)

**Comment by FatBoyXPC** [comment]
I agree with @dantleech. The docblocks could end up incorrect and you get the actual benefit of type safety with the typehinting. Remove the docblocks :)


I opened this issue because sometimes I'm in a project we can't safely add a typehint yet, so I have to rely on docblocks.

**Comment by elythyr** [comment]
> I'm in a project we can't safely add a typehint yet

I totaly get you! :'(
Even worse at work they build a code fixer rule to ensure you can't type hint in our enties.............
